src/test_helpers/integration.cljs

```diff
@@ -59,4 +78,143 @@
+          (rf/add-post-event-callback
+           cb-id
+           (fn [[event-id & _]]
+             (when-let [idx (collection/first-index #(= % event-id) @waiting-ids)]
+               ;; All `event-ids` should be processed in their original order.
+               (if (zero? idx)
+                 (do
+                   (swap! waiting-ids rest)
+                   ;; When there's nothing else to wait for, clean up resources.
+                   (when (empty? @waiting-ids)
+                     (js/clearTimeout timer-id)
+                     (rf/remove-post-event-callback cb-id)
+                     (promise-resolve)))
+                 (do
+                   (js/clearTimeout timer-id)
+                   (rf/remove-post-event-callback cb-id)
+                   (promise-reject (ex-info "event happened in unexpected order"
+                                            {:event-ids   event-ids
+                                             :waiting-for @waiting-ids}
+                                            ::out-of-order-event-id)))))))))))))
+
+(defn setup-app
+  []
+  (legacy-test/init!)
+  (if (app-initialized)
+    (p/resolved ::app-initialized)
+    (do
+      (rf/dispatch [:app-started])
+      (wait-for [:profile/get-profiles-overview-success]))))
+
+(defn setup-account
+  []
+  (if (messenger-started)
+    (p/resolved ::messenger-started)
+    (do
+      (create-multiaccount!)
+      (-> (wait-for [:messenger-started])
+          (.then #(assert-messenger-started))))))
+
+(defn integration-test
+  "Runs `f` inside `cljs.test/async` macro in a restorable re-frame checkpoint.
+
+  Option `fail-fast?`, when truthy (defaults to true), will force the test
+  runner to terminate on any test failure. Setting it to false can be useful
+  during development when you want the rest of the test suite to run due to a
+  flaky test. Prefer to fail fast in the CI to save on time & resources.
+
+  When `fail-fast?` is falsey, re-frame's state is automatically restored after
+  a test failure, so that the next integration test can run from a pristine
+  state.
+
+  Option `timeout-ms` controls the total time allowed to run `f`. The value
+  should be high enough to account for some variability, otherwise the test may
+  fail more often.
+  "
+  ([test-name f]
+   (integration-test test-name
+                     {:fail-fast? true
+                      :timeout-ms default-integration-test-timeout-ms}
+                     f))
+  ([test-name {:keys [fail-fast? timeout-ms]} f]
+   (test/async
+     done
+     (let [restore-fn (rf/make-restore-fn)]
+       (log-headline test-name)
+       (-> (p/do (f done))
+           (p/timeout timeout-ms)
+           (p/catch (fn [error]
+                      (is (nil? error))
+                      (when fail-fast?
+                        (js/process.exit 1))))
+           (p/finally (fn []
+                        (restore-fn)
+                        (done))))))))
+
+;;;; Fixtures
+
+(defn fixture-logged
```

Comment:
small nit comment: Reading `fixture-logged` makes me think this might have something to do with logging of events or something. Perhaps this could be `fixture-login-logout` or `fixture-session`? Thoughts?

Reply:
I like your `fixture-session` suggestion. Thanks @seanstrom